packages/worker/src/worker-tuner.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  FixedSizeSlotSupplier,
+  ResourceBasedTunerOptions,
+  WorkerTuner as NativeWorkerTuner,
+  SlotSupplier as NativeSlotSupplier,
+} from '@temporalio/core-bridge';
+import { Duration, msToNumber } from '@temporalio/common/lib/time';
+
+export type WorkerTuner = ResourceBasedTuner | TunerHolder;
+
+type ResourceBasedSlotsForType = ResourceBasedSlotOptions & {
+  tunerOptions: ResourceBasedTunerOptions;
+};
+
+/**
+ * @experimental
+ *
+ * Controls how slots are handed out for a specific task type.
+ *
+ * For now, only {@link ResourceBasedSlotOptions} and {@link FixedSizeSlotSupplier} are supported,
+ * but we may add support for custom tuners in the future.
+ */
+export type SlotSupplier = ResourceBasedSlotsForType | FixedSizeSlotSupplier;
+
+/**
+ * @experimental
+ *
+ * Options for a specific slot type within a {@link ResourceBasedSlotsForType}
+ */
+export interface ResourceBasedSlotOptions {
+  // Amount of slots that will be issued regardless of any other checks
+  minimumSlots: number;
+  // Maximum amount of slots permitted
+  maximumSlots: number;
+  // Minimum time we will wait (after passing the minimum slots number) between handing out new
+  // slots
+  rampThrottle: Duration;
+}
+
+/**
+ * @experimental
+ *
+ * This tuner attempts to maintain certain levels of resource usage when under load. You do not
+ * need more than one instance of this when using it for multiple slot types.
+ */
+export interface ResourceBasedTuner {
+  // Options for the tuner
+  tunerOptions: ResourceBasedTunerOptions;
+  // Options for workflow task slots. Defaults to a minimum of 2 slots and a maximum of 1000 slots
+  // with no ramp throttle
+  workflowTaskSlotOptions?: ResourceBasedSlotOptions;
+  // Options for activity task slots. Defaults to a minimum of 1 slots and a maximum of 2000 slots
+  // with 50ms ramp throttle
+  activityTaskSlotOptions?: ResourceBasedSlotOptions;
+  // Options for local activity task slots. Defaults to a minimum of 1 slots and a maximum of 2000
```

Comment:
I know this might a bit tricky to implement, but I get the feeling that if `localActivityTaskSlotSupplier` is not specified, then both regular and activities and local activities should be controlled by a single tuner…

Reply:
Yeah, I could see that. It's not hard to implement if we do it in Core. So I could do it that way.

If I make that change though I also might change how the options are passed through generally to make things a bit easier. Let me know if you have thoughts on passing through just some known option sets rather than constructing the whole tuner in the bridge.